### ISSUE TITLE
discard-groups feature not working as expected

### ISSUE THREAD

**Issue by nickjmv** [getsentry/self-hosted#2403]
### Self-Hosted Version

23.8.0

### CPU Architecture

x86_64

### Docker Version

23.0.4

### Docker Compose Version

2.17.2

### Steps to Reproduce

Deploy sentry with the following option set to true in the sentry.conf.py file:
```
SENTRY_FEATURES['projects:discard-groups'] = True
```

### Expected Result

When opening a Sentry issue, I should be able to click on the 3 dots at the top and select 'Delete and discard future events'.

### Actual Result

Half of the time when I click on this button, Sentry says that I need to add the line in my above response to the configuration, which is already there.

**Comment by hubertdeng123** [comment]
Thanks for bringing this up. I'll ask internally to see what might be the issue here

**Comment by hubertdeng123** [comment]
Is there any information that might be helpful in reproducing this?

**Comment by nickjmv** [comment]
@hubertdeng123, I can only tell you what I did to configure it. In the default configuration file of Sentry. The 'discard-groups' option is already in the big for loop:
```
############
# Features #
############

SENTRY_FEATURES["projects:sample-events"] = False
SENTRY_FEATURES.update(
    {
        feature: True
        for feature in (
        "organizations:discover",
        "organizations:events",
        "organizations:global-views",
        "organizations:incidents",
        "organizations:integrations-issue-basic",
        "organizations:integrations-issue-sync",
        "organizations:invite-members",
        "organizations:metric-alert-builder-aggregate",
        "organizations:sso-basic",
        "organizations:sso-rippling",
        "organizations:sso-saml2",
        "organizations:performance-view",
        "organizations:advanced-search",
        "projects:custom-inbound-filters",
        "projects:data-forwarding",
        "projects:discard-groups",
        "projects:plugins",
        "projects:rate-limits",
        "projects:servicehooks",
    )
    }
)
```

Nevertheless, this config was not working in the first place. Then I decided to add the line explicitly at the top like this:
```
############
# Features #
############

SENTRY_FEATURES["projects:sample-events"] = False
SENTRY_FEATURES['projects:discard-groups'] = True
SENTRY_FEATURES.update(
    {
        feature: True
        for feature in (
        "organizations:discover",
        "organizations:events",
        "organizations:global-views",
        "organizations:incidents",
        "organizations:integrations-issue-basic",
        "organizations:integrations-issue-sync",
        "organizations:invite-members",
        "organizations:metric-alert-builder-aggregate",
        "organizations:sso-basic",
        "organizations:sso-rippling",
        "organizations:sso-saml2",
        "organizations:performance-view",
        "organizations:advanced-search",
        "projects:custom-inbound-filters",
        "projects:data-forwarding",
        "projects:plugins",
        "projects:rate-limits",
        "projects:servicehooks",
    )
    }
)
```

But this gives the same result as what I described in my initial issue.

**Comment by hubertdeng123** [comment]
And you say that this happens half the time? And the other half the events are discarded properly? We run the same setup and are not seeing that issue

**Comment by nickjmv** [comment]
Yes, sometimes the button is not clickable and the message to add the line into the configuration is displayed. The other times the button just works.

**Comment by kristofclaes** [comment]
I'm a colleague of Nick and a lot of the people in my team are experiencing this issue, so I thought I'd chime in.

It's pretty strange. For me personally, the "Delete and discard future events" seems to work all the time. For other people, some with the same set of permissions, some with permissions (admin vs owner), it never works. When they click it, they get this message:

<img width="618" alt="image" src="https://github.com/getsentry/self-hosted/assets/588844/1218b21e-c7c3-4466-b976-d7f37917dad9">

When Nick did what he described in https://github.com/getsentry/self-hosted/issues/2403#issuecomment-1732941104, it seemed to work for him and other people, but a short while later it had stopped working, and people got the above popup again. ~For some reason it is still working fine for me.~ I've just checked to confirm and now it isn't working for me anymore. I also get the message that "This feature is not enabled on your Sentry installation." But we've definitely had moments where I (admin) could delete and discard future events, while others (admin and owner) in the same team for the same issue could not.

**Comment by hubertdeng123** [comment]
> It's pretty strange. For me personally, the "Delete and discard future events" seems to work all the time. For other people, some with the same set of permissions, some with permissions (admin vs owner), it never works. When they click it, they get this message:

Ok, that's super weird to me too. This feature flag looks to be from a super long time ago and I wonder if it can even be deleted. I'm going to transfer this over the the sentry repo to see if they have anything to say.
